utils_cv/detection/model.py

```diff
@@ -0,0 +1,283 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+from typing import List, Tuple, Union
+from pathlib import Path
+from functools import partial
+
+from PIL import Image
+import cv2
+import torch
+import torch.nn as nn
+from torchvision import transforms
+from torchvision.models.detection import fasterrcnn_resnet50_fpn
+from torchvision.models.detection.faster_rcnn import FastRCNNPredictor
+from torch.utils.data import Dataset, DataLoader
+import matplotlib.pyplot as plt
+
+from .references.transforms import RandomHorizontalFlip, Compose, ToTensor
+from .references.engine import train_one_epoch, evaluate
+from .references.utils import collate_fn
+from .references.coco_eval import CocoEvaluator
+from .plot import BoundingBoxParams, plot_boxes, plot_grid
+
+
+def get_bounding_boxes(
+    pred: List[dict], threshold: int = 0.6
+) -> Tuple[List[int], List[List[int]]]:
+    """ Gets the bounding boxes and labels from a prediction given a threshold.
+
+    Args:
+        pred: the output of passing in an image to torchvision's FasterRCNN
+        model
+        threshold: the minimum threshold to accept.
+
+    Return:
+        a list of labels and bounding boxes that pass the minimum threshold.
+    """
+    pred_labels = list(pred[0]["labels"].cpu().numpy())
+    pred_boxes = list(pred[0]["boxes"].detach().cpu().numpy().astype(np.int32))
+    pred_scores = list(pred[0]["scores"].cpu().numpy())
+
+    qualified_labels = []
+    qualified_boxes = []
+    for label, box, score in zip(pred_labels, pred_boxes, pred_scores):
+        if score > threshold:
+            qualified_labels.append(label)
+            qualified_boxes.append(box)
+    return qualified_labels, qualified_boxes
+
+
+def get_transform(train: bool) -> List[object]:
```

Comment:
this is a snippet from the torchvision tutorial: https://pytorch.org/tutorials/intermediate/torchvision_tutorial.html#writing-a-custom-dataset-for-pennfudan

Reply:
Maybe this function should be in dataset? It's only used there atm